code/modules/mod/modules/modules_science.dm

```diff
@@ -159,3 +159,167 @@
 	if(!attached_module.mod?.wearer)
 		return
 	wearer.set_output(attached_module.mod.wearer)
+
+///Anomaly Locked - Causes the module to not function without an anomaly.
+/obj/item/mod/module/anomaly_locked
+	name = "MOD anomaly locked module"
+	desc = "A form of a module, locked behind an anomalous core to function."
+	incompatible_modules = list(/obj/item/mod/module/anomaly_locked)
+	/// The core item the module runs off.
+	var/obj/item/assembly/signaler/anomaly/core
+	/// Accepted types of anomaly cores.
+	var/list/accepted_anomalies = list(/obj/item/assembly/signaler/anomaly)
+	/// If this one starts with a core in.
+	var/prebuilt = FALSE
+
+/obj/item/mod/module/anomaly_locked/Initialize(mapload)
+	. = ..()
+	if(!prebuilt || !length(accepted_anomalies))
+		return
+	var/core_path = pick(accepted_anomalies)
+	core = new core_path(src)
+	update_icon_state()
+
+/obj/item/mod/module/anomaly_locked/Destroy()
+	QDEL_NULL(core)
+	return ..()
+
+/obj/item/mod/module/anomaly_locked/examine(mob/user)
+	. = ..()
+	if(!length(accepted_anomalies))
+		return
+	if(core)
+		. += span_notice("There is a [core.name] installed in it. You could remove it with a <b>screwdriver</b>...")
+	else
+		var/core_string = ""
+		for(var/path in accepted_anomalies)
+			var/atom/core_path = path
+			if(path == accepted_anomalies[1])
+				core_string += initial(core_path.name)
+			else if(path == accepted_anomalies[length(accepted_anomalies)])
+				core_string += "or [initial(core_path.name)]"
+			else
+				core_string += ", [initial(core_path.name)]"
```

Comment:
Seems like this could be replaced with the proc `english_list()` ?

Construct a list of core names, then pass it to `english_list()` - specify `and_text` to be ` or `